src/cryptonote_core/miner.h

```diff
@@ -119,8 +146,24 @@ namespace cryptonote
     bool m_do_print_hashrate;
     bool m_do_mining;
 
+    // background mining stuffs ..
+
+    bool set_is_background_mining_enabled(bool is_background_mining_enabled);
+    bool background_worker_thread();
+    std::atomic<bool> m_is_background_mining_enabled;
+    boost::mutex m_is_background_mining_enabled_mutex;
+    boost::condition_variable m_is_background_mining_enabled_cond;
+    std::atomic<bool> m_is_background_mining_started;
+    boost::mutex m_is_background_mining_started_mutex;
+    boost::condition_variable m_is_background_mining_started_cond;    
+    boost::thread m_background_mining_thread;
+    uint64_t m_min_idle_seconds;
+    uint8_t m_idle_threshold;
+    uint8_t m_mining_target;
+    std::atomic<uint64_t> m_miner_extra_sleep;
+    static bool get_system_times(uint64_t& total_time, uint64_t& idle_time);
+    static bool get_process_time(uint64_t& total_time);
+    static uint8_t get_percent_of_total(uint64_t some_time, uint64_t total_time);
+    static bool ac_line_status(); // true = plugged in, else nope
```

Comment:
I'm not either. What does "line" mean here ? Is it a typo ? Did you mean "ac_live_status" ?